srcalc/src/main/java/gov/va/med/srcalc/util/RetrievalEnum.java

```diff
@@ -1,77 +1,131 @@
-				final Variable variable, final String key)
-		{
-			if(patient.getBmiDate() != null)
-			{
-				variableEntry.getDynamicValues().put(key, String.valueOf(patient.getBmi()));
-				variable.setRetrievalDateString(VariableEntry.makeRetrievalMessage(patient.getBmiDate()));
-			}
-		}
-	}, WEIGHT {
-		@Override
-		public void execute(final Patient patient, final VariableEntry variableEntry,
-				final Variable variable, final String key)
-		{
-			if(patient.getWeightDate() != null)
-			{
-				variableEntry.getDynamicValues().put(key, String.valueOf(patient.getWeight()));
-				variable.setRetrievalDateString(VariableEntry.makeRetrievalMessage(patient.getWeightDate()));
-			}
-		}
-	}, WEIGHT_6_MONTHS_AGO {
-		@Override
-		public void execute(final Patient patient, final VariableEntry variableEntry,
-				final Variable variable, final String key)
-		{
-			if(patient.getWeight6MonthsAgoDate() != null)
-			{
-				variableEntry.getDynamicValues().put(key, String.valueOf(patient.getWeight6MonthsAgo()));
-				variable.setRetrievalDateString(VariableEntry.makeRetrievalMessage(patient.getWeight6MonthsAgoDate()));
-			}
-		}
-	}, HEIGHT {
-		@Override
-		public void execute(final Patient patient, final VariableEntry variableEntry,
-				final Variable variable, final String key)
-		{
-			if(patient.getHeightDate() != null)
-			{
-				variableEntry.getDynamicValues().put(key, String.valueOf(patient.getHeight()));
-				variable.setRetrievalDateString(VariableEntry.makeRetrievalMessage(patient.getHeightDate()));
-			}
-		}
-	}, CARDIAC_AGE {
-		@Override
-		public void execute(final Patient patient, final VariableEntry variableEntry,
-				final Variable variable, final String key)
-		{
-			variableEntry.getDynamicValues().put(key, String.valueOf(patient.getAge()));
-		}
-	};
-	
-	public abstract void execute(final Patient patient, final VariableEntry variableEntry,
-			final Variable variable, final String key);
+public enum RetrievalEnum
+{
+    GENDER
+    {
+        @Override
+        public void execute(final Patient patient, final VariableEntry variableEntry, final Variable variable,
+                final String key)
+        {
+            variableEntry.getDynamicValues().put(key, patient.getGender());
+        }
+    },
+    AGE
+    {
+        @Override
+        public void execute(final Patient patient, final VariableEntry variableEntry, final Variable variable,
+                final String key)
+        {
+            variableEntry.getDynamicValues().put(key, String.valueOf(patient.getAge()));
+        }
+    },
+    BMI
+    {
+        @Override
+        public void execute(final Patient patient, final VariableEntry variableEntry, final Variable variable,
+                final String key)
+        {
+            if (patient.getBmi() != null)
+            {
+                variableEntry.getDynamicValues().put(key, String.valueOf(patient.getBmi().getValue()));
+                variable.setRetrievalDateString(makeRetrievalString(
```

Comment:
Didn't we decide on moving these strings to the `VariableEntry` object instead of `AbstractVariable`?